src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -229,17 +229,18 @@ private void drawLineWithTranslate(Graphics2D graphics, Line2D line, double xOff
    * @return The "real" Point
    */
   private static Point getPointOnMap(Point point, int xOffset, int yOffset, int width, int height) {
-    double x = point.getX();
-    double y = point.getY();
+    int x = (int) point.getX();
+    int y = (int) point.getY();
     if (x - width > xOffset) {
-      point.setLocation(x - width, y);
+      point = new Point(x - width, y);
```

Comment:
@DanVanAtta I actually personally don't like adding the final keyword in the parameter section, since it makes the code more difficult to read IMO.
I though it would be better to only modify this point, so that there's no need in creating a new local field